src/ann_solo/spectrum_similarity.py

```diff
@@ -0,0 +1,741 @@
+import math
+import warnings
+from typing import Optional
+
+import numpy as np
+import scipy.spatial.distance
+import scipy.special
+import scipy.stats
+
+from ann_solo import spectrum
+
+
+class SpectrumSimilarityCalculator:
+    def __init__(
+        self, ssm: spectrum.SpectrumSpectrumMatch, top: Optional[int] = None
+    ):
+        """
+        Instantiate the `SpectrumSimilarityFactory` to compute various spectrum
```

Comment:
**question:** Did you mean `SpectrumSimilarityCalculator` here? If not, I wouldn't really call this a true factory pattern, although that's pretty nitpicky. Feel free to ignore!

Reply:
Thanks. This was lingering from a previous naming. Because it's indeed not really a factory pattern, I switched to `SpectrumSimilarityCalculator` from the initial `SpectrumSimilarityFactory`. The former is a better name, but the refactoring didn't catch the docstring apparently.